examples/wait-for-domain-example.c

```diff
@@ -0,0 +1,88 @@
+/* The LibVMI Library is an introspection library that simplifies access to
+ * memory in a target virtual machine or in a file containing a dump of
+ * a system's physical memory.  LibVMI is based on the XenAccess Library.
+ *
+ * Author: Alexandru Isaila (aisaila@bitdefender.com)
+ *
+ * This file is part of LibVMI.
+ *
+ * LibVMI is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * LibVMI is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with LibVMI.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdlib.h>
+#include <signal.h>
+#include <libvmi/libvmi.h>
+#include <libvmi/libvmi.h>
+#include <libvmi/events.h>
+
+static int interrupted = 0;
+static void close_handler(int sig)
+{
+    interrupted = sig;
+}
+vmi_event_t interrupt_event = {0};
+
+event_response_t wait_for_domain(vmi_instance_t vmi, vmi_event_t *event)
+{
+    watch_domain_data_t *data = (watch_domain_data_t *)event->data;
```

Comment:
event->data is user defined, you can't just use it like this. You have to create an event structure similar to the existing event types to convey extra information about the event